src/components/SingletonEditor.test.tsx

```diff
@@ -46,21 +46,21 @@ describe(Editor.name, () => {
     const appServices = defaultAppServices as AppServices;
 
     const DemoComponent = () => {
-      const { setContent, unsetContent } = useSingletonEditor();
+      // const { setContent, unsetContent } = useSingletonEditor();
```

Comment:
Creo que este test (_should show and hide EmailEditor when design is set and unset_) se podría eliminar o tal vez modificar por algo así:

```js
const TestComponente = ({ showEditor }) => {
  if (!showEditor) {
    return <p>No se muestra editor</p>
  }

  useSingletonEditor({
    initialContent: {}
  });

  return <p>No se muestra editor</p>
};
```

Cuando renderizemos `<TestComponente showEditor=true>` se debería mostrar el editor, cuando cambiemos `showEditor` a false, se debería ocultar.

Reply:
Lo intente pero queda en un bucle infinito

Reply:
Mmm, eso es preocupante, porque podría ser haya algún problema que no estamos viendo en la UI.

Voy a intentarlo, luego te cuento.

Reply:
Ya descubrí que sucede con ese bucle infinito, en useSingletonEditor tenemos como dependencia initialContent y setContent ambos están forzando el renderizado del componente cada que el hook setea el contenido se renderiza

Reply:
Y gracias a `react-query` eso no es un bucle infinito ni se hacen múltiples queries, no?

Acá armé el test con un `initialContent` que no muta y efectivamente funciona como es esperado:

<img width="707" alt="image" src="https://user-images.githubusercontent.com/1157864/164708902-a4d31d16-84a4-48b8-a97b-db10cbbda3d1.png">

Ahora, ¿cual es la solución? ¿evitar que el componente que llama al hook se renderize más de una vez? ¿hacer que el hook sea tolerante a eso? ¿ambas?

Reply:
Igual, estoy viendo que aún no lo entendí del todo. ¿Que es lo que hace que `<DemoComponentWithEditor>` se rederize nuevamente?

Cuando puedas, lo hablamos.

Reply:
Hacer que el hook sea tolerante a eso, por ejemplo en esta línea no deberíamos escuchar los cambios del initialContent y tampoco los cambios del setContent https://github.com/FromDoppler/doppler-editors-webapp/blob/e2c2482bb082d97fda5ad0878d2816c241b9f8b5/src/components/SingletonEditor.tsx#L67

Reply:
Creo que eso no funcionaría bien, porque cuando cargamos otro template, eso debería volver a ejecutarse.

Reply:
Encontramos que `useSingletonEditor()` generaba renderizados infinitos al tener como dependencia el _initialContent_ y ser responsable de modificar el contenido _"setContent()"_, para solucionar este comportamiento implementamos un array de dependencia que es pasado por quien usa el `useSingletonEditor(config, deps)` de esta forma trasladamos la responsabilidad de renderizar el contenido a quien configura el editor.